04-data-tree/index.js

```diff
@@ -0,0 +1,40 @@
+class Node {
+  constructor(value) {
+    this.value = value;
+    this.children = [];
+  }
+
+  addChild(child) {
+    this.children.push(child);
+  }
+  find(value) {
+    for (let i = 0; i < this.children.length; i++) {
+      if (this.children[i].value == value) {
+        return this.children[i];
+      } else if (this.children[i] == this.children[this.children.length - 1]) {
```

Comment:
What does this condition check?

Reply:
this one else if (this.children[i] == this.children[this.children.length - 1]) ??

Reply:
yes

Reply:
if  it iterate  and doesnt find the value then starts this instruction ( iterate form beginning but level lower)